scripts/example_script.py

```diff
@@ -0,0 +1,241 @@
+recurse_get_resource(gc, collection_id, 'item', item_list,
+                     parent_is_collection=True)
+
+temp_img_id = None
+print('Listing all %s items in the collection %s' % (img_type, collection_name))
+for item in item_list:
+    if img_type in item['name']:
+        temp_img_id = item['_id']
+        print('image name %s ' % (item['name']))
+        img_list.append(temp_img_id)
+        turn_into_a_large_image(gc, temp_img_id)
+
+# now create PATIENT folders inside our new folder
+
+patient_folder_list = []
+patient_folder_name = 'Patient%d'
+
+for i in range(1, 4):
+    temp_name = patient_folder_name % (i)
+    try:
+        patient_folder_list.append(gc.load_or_create_folder(temp_name,
+                                                            new_folder_id,
+                                                            'folder'))
+    except HttpError as err:
+        print_http_error(err)
+        print ('exited')
+        exit()
+
+# now lets move an image into the PATIENT1 folder
+
+patient1_folder = patient_folder_list[0]
+patient1_folder_id = patient1_folder['_id']
+
+src_img_id = img_list[0]
+
+move_params = {
+    'resources': json.dumps({
+        'item': [str(src_img_id)]}),
+    'parentType': 'folder',
+    'parentId': patient1_folder_id
+}
+
+try:
+    gc.put('resource/move', move_params)
+except HttpError as err:
+    print_http_error(err)
+    print ('exited')
+    exit()
+
+# now lets edit the PATIENT1 folder
+
+patient1_meta_data = {'PatientAge': 22, 'PatientDeceased': True}
+
+try:
+    gc.addMetadataToFolder(patient1_folder_id, patient1_meta_data)
+except HttpError as err:
+    print_http_error(err)
+    print ('exited')
+    exit()
+
+# lets add meta data to the image we moved
+# note how the id remains the same
+new_src_img_meta_data = {'SlideSource': 'FromTCGA', 'SlideType': 'DX'}
+try:
+    gc.addMetadataToItem(src_img_id, new_src_img_meta_data)
+except HttpError as err:
+    print_http_error(err)
+    print ('exited')
+    exit()
+
+# find all folders determine if whether each folder is empty or not
+all_folders = []
+
+recurse_get_resource(gc, collection_id, 'folder',
+                     all_folders, parent_is_collection=True)
+
+# if the folder size is zero then delete it
+for folder in all_folders:
+
+    if 'size' in folder and isEmpty(gc, folder['_id']):
+        try:
+            print('will delete folder %s' % folder['name'])
+
+            gc.delete('folder/%s' % folder['_id'])
+
+        except HttpError as err:
+            print_http_error(err)
+            print('could not delete folder %s' % (folder['name']))
+            print ('exited')
+            exit()
+# already have a list of all items under the designated collection
+# check whether a file under an item is a README.txt
+# if so delete the entire item
+for item in item_list:
+    temp_files = gc.listFile(item['_id'])
+    for girder_file in temp_files:
+        if 'README.txt' in girder_file['name']:
```

Comment:
We should compare this case-insensitively, but we can look for an exact match otherwise.